src/fuzz/spanner_emulator_ddl_statement_proto_to_string_test.cc

```diff
@@ -0,0 +1,385 @@
+    // test that the max wraps back to 1
+    column_data_type.set_length(2621440);
+    EXPECT_EQ(toString(column_data_type.scalartype(), column_data_type.length(), 
+        column_data_type.lengthtype()), "STRING( 1 )");
+
+    column_data_type.set_length(-500);
+    column_data_type.set_lengthtype(ColumnDataType::UNBOUND);
+    column_data_type.set_scalartype(ColumnDataType::BYTES);
+    EXPECT_EQ(toString(column_data_type.scalartype(), column_data_type.length(), 
+        column_data_type.lengthtype()), "BYTES( -500 )");
+    column_data_type.set_lengthtype(ColumnDataType::MAX);
+    EXPECT_EQ(toString(column_data_type.scalartype(), column_data_type.length(), 
+        column_data_type.lengthtype()), "BYTES( MAX )");
+    column_data_type.set_lengthtype(ColumnDataType::BOUND);
+    EXPECT_EQ(toString(column_data_type.scalartype(), column_data_type.length(), 
+        column_data_type.lengthtype()), "BYTES( 501 )");
+    // test that the max wraps back to 1
+    column_data_type.set_length(10485760);
+    EXPECT_EQ(toString(column_data_type.scalartype(), column_data_type.length(), 
+        column_data_type.lengthtype()), "BYTES( 1 )");
+}
+
+TEST(DDLStatementProtoToString, IsColumnNotNullToStringTest) {
+    EXPECT_EQ(isColumnNotNullToString(true), "NOT NULL");
+    EXPECT_EQ(isColumnNotNullToString(false), "");
+}
+
+TEST(DDLStatementProtoToString, ColumnOptionsToStringTest) {
+    EXPECT_EQ(columnOptionsToString(false), "OPTIONS ( allow_commit_timestamp = null )");
+    EXPECT_EQ(columnOptionsToString(true), "OPTIONS ( allow_commit_timestamp = true )");
+}
+
+TEST(DDLStatementProtoToString, ToPrimaryKeysTest) {
+    CreateTable create_table;
+
+    Column* column1 = create_table.add_primarykeys();
+    EXPECT_EQ(column1->has_columnname(), false);
+    EXPECT_EQ(column1->has_orientation(), false);
+
+    column1->set_orientation(Column::ASC);
+    column1->set_columnname("testColumn1");
+
+    Column* column2 = create_table.add_primarykeys();
+    EXPECT_EQ(column2->has_columnname(), false);
+    EXPECT_EQ(column2->has_orientation(), false);
+
+    column2->set_orientation(Column::DESC);
+    column2->set_columnname("testColumn2");
+
+    EXPECT_EQ(create_table.primarykeys().size(), 2);
+
+    EXPECT_EQ(toPrimaryKeys(create_table.primarykeys()), "testColumn1 ASC,testColumn2 DESC");
+}
+
+TEST(DDLStatementProtoToString, ColumnToPrimaryKeyTest) {
+    Column column;
+    EXPECT_EQ(column.has_columnname(), false);
+    EXPECT_EQ(column.has_orientation(), false);
+
+    column.set_orientation(Column::ASC);
+    column.set_columnname("testColumn");
+
+    EXPECT_EQ(columnToPrimaryKey(column), "testColumn ASC");
+}
+
+TEST(DDLStatementProtoToString, ColumnOrientationTest) {
+    Column column;
+    EXPECT_EQ(column.has_orientation(), false);
+
+    column.set_orientation(Column::ASC);
+    EXPECT_EQ(toString(column.orientation()), "ASC");
+
+    column.set_orientation(Column::DESC);
+    EXPECT_EQ(toString(column.orientation()), "DESC");
+}
```

Comment:
Maybe add a test that mirrors the fuzz test but validates that we create the database correctly for a well formed statement?

Reply:
Adding as a TODO due to build issues with testing this